classes/local/factor/object_factor_base.php

```diff
@@ -281,7 +281,17 @@ public function revoke_user_factor($factorid = null) {
         global $DB, $USER;
 
         if (!empty($factorid)) {
+            // If we have an explicit factor id, this means we need to be careful about the user.
             $params = ['id' => $factorid];
+            $existing = $DB->get_record('tool_mfa', $params);
+            if (empty($existing)) {
+                return false;
+            }
+            $matchinguser = $existing->userid == $USER->id;
+            if (!(is_siteadmin() || $matchinguser)) {
+                // We aren't admin, and this isn't our factor.
```

Comment:
Perhaps it might be easier to skim through if the condition was
```
if (!is_siteadmin() && !$matchinguser) {
```
 (e.g. less OR and !() conditions)

Reply:
+1 for this one. Took me few seconds to process` !(is_siteadmin() || $matchinguser)`

Reply:
Im in the opposite camp, purely because "not siteadmin or user" sounds more natural in my brain than "not siteadmin and not user". Im happy to change it, but I find the current easier to grok.

Reply:
OK ill change it